code/game/turfs/liquid_turfs.dm

```diff
@@ -28,54 +33,25 @@
-		if(mob_liquid_depth != old_turf.mob_liquid_depth)
-			animate(arrived_mob, pixel_y = arrived_mob.pixel_y + mob_liquid_depth - old_turf.mob_liquid_depth, time = arrived_mob.cached_multiplicative_slowdown + arrived_mob.next_move_slowdown, flags = ANIMATION_PARALLEL)
-	else
-		//The mask is spawned below the mob, then the animate() raises it up, giving the illusion of dropping into water, combining with the animate to actual drop the pixel_y into the water
-		arrived_mob.add_filter(MOB_LIQUID_TURF_MASK, 1, alpha_mask_filter(0, height_to_use - MOB_LIQUID_TURF_MASK_HEIGHT, icon('icons/turf/alpha_64.dmi', "liquid_alpha"), null, MASK_INVERSE))
-
-		animate(arrived_mob.get_filter(MOB_LIQUID_TURF_MASK), y = height_to_use - (MOB_LIQUID_TURF_MASK_HEIGHT - mob_liquid_height), time = arrived_mob.cached_multiplicative_slowdown + arrived_mob.next_move_slowdown)
-		animate(arrived_mob, pixel_y = arrived_mob.pixel_y + mob_liquid_depth, time = arrived_mob.cached_multiplicative_slowdown + arrived_mob.next_move_slowdown, flags = ANIMATION_PARALLEL)
-
-	if(!arrived_mob.throwing)
-		arrived_mob.next_move_slowdown += (arrived_mob.get_liquid_slowdown() * slowdown_multiplier)
-
-/turf/open/liquid/Exited(atom/movable/leaver, direction)
+/turf/open/liquid/get_submerge_height(turf_only = FALSE)
 	. = ..()
-	if(!ismob(leaver))
+	if(SEND_SIGNAL(src, COMSIG_TURF_CHECK_COVERED))
 		return
-	var/mob/mob_leaver = leaver
-	if(!mob_leaver.get_filter(MOB_LIQUID_TURF_MASK))
+	if(length(canSmoothWith) && !CHECK_MULTIPLE_BITFIELDS(smoothing_junction, (SOUTH_JUNCTION|EAST_JUNCTION|WEST_JUNCTION)))
 		return
+	. += mob_liquid_height
 
-	var/turf/open/liquid/new_turf = mob_leaver.loc
-	if(istype(new_turf))
-		if(length(new_turf.canSmoothWith))
-			if(!SEND_SIGNAL(new_turf, COMSIG_TURF_CHECK_COVERED) && CHECK_MULTIPLE_BITFIELDS(new_turf.smoothing_junction, (SOUTH_JUNCTION|EAST_JUNCTION|WEST_JUNCTION)))
-				return
-		else if(!SEND_SIGNAL(new_turf, COMSIG_TURF_CHECK_COVERED))
-			return
-
-	var/icon/mob_icon = icon(mob_leaver.icon)
-	animate(mob_leaver.get_filter(MOB_LIQUID_TURF_MASK), y = ((64 - mob_icon.Height()) * 0.5) - MOB_LIQUID_TURF_MASK_HEIGHT, time = mob_leaver.cached_multiplicative_slowdown + mob_leaver.next_move_slowdown)
-	animate(mob_leaver, pixel_y = mob_leaver.pixel_y - mob_liquid_depth, time = mob_leaver.cached_multiplicative_slowdown + mob_leaver.next_move_slowdown, flags = ANIMATION_PARALLEL)
-	addtimer(CALLBACK(mob_leaver, TYPE_PROC_REF(/atom, remove_filter), MOB_LIQUID_TURF_MASK), mob_leaver.cached_multiplicative_slowdown + mob_leaver.next_move_slowdown)
+/turf/open/liquid/get_submerge_depth()
+	if(SEND_SIGNAL(src, COMSIG_TURF_CHECK_COVERED))
+		return 0
+	if(length(canSmoothWith) && !CHECK_MULTIPLE_BITFIELDS(smoothing_junction, (SOUTH_JUNCTION|EAST_JUNCTION|WEST_JUNCTION)))
```

Comment:
Do you not submerge at shores or what's up with this?

Reply:
This is just a visual thing. On these dirs, you'd be submerged in sand which looks goofy.